### ISSUE TITLE
BFF crashes

### ISSUE THREAD

**Issue by baritonomarchetto** [njz3/vJoyIOFeederWithFFB#12]
I have already installed BFF on my win10 laptop for the very first tests with mame and a CH340 MEGA. That far, all good.
In order to test model 2 and supermodel I have set up a desktop PC with a dedicated videocard.
The new PC showed some difficulty with the CH340 MEGA clone I was using, suggesting some serial communication issue I have not resolved yet. I have then switched to a 16u2 MEGA 2560. This compiles ok, but when I launch BFF GUI, it crashes after recognizing the board. 
Does BFF compile a log file of some sort in order to see what happens?
Notice that the both old and new MEGAs works in combination with BFF GUI on another PC (my laptop). No crashes at all there.

**Comment by njz3** [comment]
If you run the software in console mode (BackForceFeeder.exe from command line) and not the GUI mode, if there is a crash it will give you an exception message that will be printed to the console.

**Comment by baritonomarchetto** [comment]
Ok, this is what happens running backforcefeeder from command line (no GUI exe):

```
C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>backforcefeeder
Starting console mode of BackForceFeeder v0.6.0.68 Made for Gamoover by njz3!
****************************************
*** PRESS ESCAPE THEN ENTER TO LEAVE ***
****************************************
IMPOR|    1.773715:[MAIN] Machine ID=C28E2582C498B5A3
IMPOR|    1.775236:[MAIN] Invalid license.dat found for user DESKTOP-G1DMDR7@mail.com
IMPOR|    1.775347:[MAIN] No valid license.dat found, please contact the author and give your email and your machine ID to register
INFOR|    1.775665:[MAIN] Running in trial mode
IMPOR|    1.779471:[MAIN] Running as standard user, trying realtime priority
IMPOR|    1.783964:[MAIN] Setting realtime priority failed, got: High
IMPOR|    1.974649:[MANAGER] Program configured for PWM_SHAKER
IMPOR|    3.230073:[MANAGER] Found io board on COM3 version=0.2.0.0 type=IO BOARD ON MEGA2560
IMPOR|    3.454254:[MANAGER] Initializing IO board
IMPOR|    3.454849:[USBSerial] Performing IO board initialization
IMPOR|    4.489779:[USBSerial] IO board initialization done
INFOR|    4.490311:[MANAGER] Configuring IO board for pwmmode=0
INFOR|    4.509096:[MANAGER] Configuring IO board for wheelmode=2
INFOR|    4.526871:[MANAGER] Configuring IO board for pedalmode=0
INFOR|    4.544824:[MANAGER] Configuring IO board for ffbcontrollermode=0

Unhandled Exception: INFOR|    4.684267:[MAMEOutputsAgent] Entering HiddenWindow Thread
System.NullReferenceException: Object reference not set to an instance of an object.
   at BackForceFeeder.Inputs.RawInput.?(Boolean )
   at BackForceFeeder.Inputs.InputsManager. (UInt64 )
   at BackForceFeeder.Inputs.InputsManager. (Int32 , Int32 )
   at BackForceFeeder.BackForceFeeder.BFFManager. ()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()

C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>
```

Can you spot what is going wrong?

(thanks!)

**Comment by njz3** [comment]
There is a bug in the input management. I must reproduce to fix it.
Can you attach your configurations and control sets files (as a zip) ?
Do you use the default arduino code ?

**Comment by baritonomarchetto** [comment]
This was not the default code, but the one with my custom inputs. I had the crash also with the stock sketch anyway, so I would bet it will return the same error.
To be 100% sure I will upload your sketch tomorrow, relaunch the feeder, cooy the error and upload the zip you asked.
Thankyou

**Comment by baritonomarchetto** [comment]
Here I am.

I have unistalled BFF via your exe, then reinstalled it. 
Uploaded to my MEGA your code (the one present in the release pack, not the one in the "open" directory).
Launched BFF (no GUI) via command prompt, this is the error I get:

```
C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>backforcefeeder
Starting console mode of BackForceFeeder v0.6.0.68 Made for Gamoover by njz3!
****************************************
*** PRESS ESCAPE THEN ENTER TO LEAVE ***
****************************************
IMPOR|    1.372740:[MAIN] Machine ID=C28E2582C498B5A3
IMPOR|    1.373920:[MAIN] Invalid license.dat found for user DESKTOP-G1DMDR7@mail.com
IMPOR|    1.374030:[MAIN] No valid license.dat found, please contact the author and give your email and your machine ID to register
INFOR|    1.374344:[MAIN] Running in trial mode
IMPOR|    1.377584:[MAIN] Running as standard user, trying realtime priority
IMPOR|    1.378210:[MAIN] Setting realtime priority failed, got: High
IMPOR|    1.385779:[MANAGER] Program configured for PWM_SHAKER
IMPOR|    2.638632:[MANAGER] Found io board on COM3 version=0.2.0.0 type=IO BOARD ON MEGA2560
IMPOR|    2.748889:[MANAGER] Initializing IO board
IMPOR|    2.749484:[USBSerial] Performing IO board initialization
IMPOR|    3.786156:[USBSerial] IO board initialization done
INFOR|    3.786699:[MANAGER] Configuring IO board for pwmmode=0
INFOR|    3.805438:[MANAGER] Configuring IO board for wheelmode=2

Unhandled Exception: INFOR|    3.823039:[MANAGER] Configuring IO board for pedalmode=0
INFOR|    3.840386:[MANAGER] Configuring IO board for ffbcontrollermode=0
INFOR|    3.909719:[MAMEOutputsAgent] Entering HiddenWindow Thread
System.NullReferenceException: Object reference not set to an instance of an object.
   at BackForceFeeder.Inputs.RawInput.?(Boolean )
   at BackForceFeeder.Inputs.InputsManager. (UInt64 )
   at BackForceFeeder.Inputs.InputsManager. (Int32 , Int32 )
   at BackForceFeeder.BackForceFeeder.BFFManager. ()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()

C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>
```
My configuration files are stored [here](https://github.com/baritonomarchetto/pic_repo) (BackForceFeeder.7z), but read the following before getting them.

I also made another test: completely deleted the backforcefeeder folders under %USERPROFILE%\Documents\BackForceFeeder , launched BFF to have the folders recreated, then launched the application. Still crash. Got this error:

```
C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>backforcefeeder
Starting console mode of BackForceFeeder v0.6.0.68 Made for Gamoover by njz3!
****************************************
*** PRESS ESCAPE THEN ENTER TO LEAVE ***
****************************************
ERROR|    0.009206:Could not load configuration file C:\Users\HOME\Documents\BackForceFeeder\Config\ApplicationCfg.json due to Could not find file 'C:\Users\HOME\Documents\BackForceFeeder\Config\ApplicationCfg.json'.
ERROR|    0.009867:Could not load configuration file C:\Users\HOME\Documents\BackForceFeeder\Config\HardwareCfg.json due to Could not find file 'C:\Users\HOME\Documents\BackForceFeeder\Config\HardwareCfg.json'.
IMPOR|    1.182064:[MAIN] Machine ID=C28E2582C498B5A3
IMPOR|    1.182709:[MAIN] Invalid license.dat found for user DESKTOP-G1DMDR7@mail.com
IMPOR|    1.182819:[MAIN] No valid license.dat found, please contact the author and give your email and your machine ID to register
INFOR|    1.183173:[MAIN] Running in trial mode
IMPOR|    1.186556:[MAIN] Running as standard user, trying realtime priority
IMPOR|    1.187182:[MAIN] Setting realtime priority failed, got: High
IMPOR|    1.194885:[MANAGER] Program configured for PWM_DIR
IMPOR|    2.443498:[MANAGER] Found io board on COM3 version=0.2.0.0 type=IO BOARD ON MEGA2560
IMPOR|    2.555212:[MANAGER] Initializing IO board
IMPOR|    2.555820:[USBSerial] Performing IO board initialization
IMPOR|    3.590935:[USBSerial] IO board initialization done
INFOR|    3.591702:[MANAGER] Configuring IO board for pwmmode=0
INFOR|    3.610484:[MANAGER] Configuring IO board for wheelmode=2
INFOR|    3.628521:[MANAGER] Configuring IO board for pedalmode=0
INFOR|    3.646529:[MANAGER] Configuring IO board for ffbcontrollermode=0

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at BackForceFeeder.Inputs.RawInput.?(Boolean )
   at BackForceFeeder.Inputs.InputsManager. (UInt64 )
   at BackForceFeeder.Inputs.InputsManager. (Int32 , Int32 )
   at BackForceFeeder.BackForceFeeder.BFFManager. ()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
INFOR|    3.721859:[MAMEOutputsAgent] Entering HiddenWindow Thread

C:\Program Files\BackForceFeeder\BackForceFeeder-v0.6.0.68>
```

**Comment by njz3** [comment]
I have reproduced the crash.
It is due to missing control set in the ControlSets directory (you don't even have a default one), which leads the software to have a null reference.
I will fix that. In the meantime, you can check by copying a default control set from another working installation.

**Comment by njz3** [comment]
Should be fixed in 0.6.0.70, please give it a try

**Comment by baritonomarchetto** [comment]
It actually is!

Thanks :)
